src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/ExperimentWorkflowNavigator.java

```diff
@@ -0,0 +1,80 @@
+package org.breedinginsight.brapps.importer.services.processors.experiment;
+
+import io.micronaut.context.annotation.Primary;
+import org.breedinginsight.brapps.importer.model.imports.ImportServiceContext;
+import org.breedinginsight.brapps.importer.model.workflow.ImportWorkflow;
+import org.breedinginsight.brapps.importer.model.workflow.ExperimentWorkflow;
+import org.breedinginsight.brapps.importer.model.workflow.ImportWorkflowResult;
+
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Primary
+@Singleton
+public class ExperimentWorkflowNavigator implements ExperimentWorkflow {
+    private final List<ExperimentWorkflow> workflows;
+
+    public ExperimentWorkflowNavigator(List<ExperimentWorkflow> workflows) {
+        this.workflows = workflows;
+    }
+
+    @Override
+    public Optional<ImportWorkflowResult> process(ImportServiceContext context) throws Exception {
+
+        // NOTE: Couldn't throw exception from map lambda
+        for (ExperimentWorkflow workflow : workflows) {
+            Optional<ImportWorkflowResult> result = workflow.process(context);
+            if (result.isPresent()) {
+                return result;
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public List<ImportWorkflow> getWorkflows() throws Exception {
+        // Each workflow returns in the field workflow the metadata about the workflow that processed the import context.
+        // Loop over all workflows, processing a null context, to collect just the metadata
```

Comment:
The `throws Exception` doesn't apply here since there isn't a scenario where getWorkflows() would ever pass up an exception from a lower level; it's only here because `throws Exception` was added to the Workflow#process signature.

Reply:
The throws Exception was added there because ExperimentWorkflowNavigator::getWorkflows calls process on the ExperimentWorkflow and the process method in CreateNewExperimentWorkflow can throw exceptions. The particular code path for returning workflows wouldn't but the method in general can. I'll look at the BI-2134 stuff.